modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java

```diff
@@ -1308,22 +1314,39 @@ private void _bindObjectDefinitions(ObjectRelationship objectRelationship)
 			_objectDefinitionPersistence.findByPrimaryKey(
 				objectRelationship.getObjectDefinitionId2());
 
-		String objectDefinition2PreviousRESTContextPath =
-			objectDefinition2.getRESTContextPath();
+		Tree tree = _treeFactory.createObjectDefinitionTree(
+			objectDefinition2.getObjectDefinitionId());
 
-		if (objectDefinition1.isApproved() == objectDefinition2.isApproved()) {
-			objectDefinition2.setRootObjectDefinitionId(
-				objectDefinition1.getObjectDefinitionId());
-		}
-		else {
-			objectDefinition2.setRootObjectDefinitionId(
-				objectDefinition2.getObjectDefinitionId());
-		}
+		Iterator<Node> iterator = tree.iterator();
+
+		while (iterator.hasNext()) {
+			Node node = iterator.next();
+
+			ObjectDefinition objectDefinition =
+				objectDefinitionLocalService.fetchObjectDefinition(
+					node.getPrimaryKey());
+
+			String objectDefinitionPreviousRESTContextPath =
+				objectDefinition.getRESTContextPath();
+
+			objectDefinition.setRootObjectDefinitionId(
+				objectDefinition1.getRootObjectDefinitionId());
```

Comment:
From Feliphe's code, there was this condition:
```
if (objectDefinition1.isApproved() == objectDefinition2.isApproved()) {
	objectDefinition2.setRootObjectDefinitionId(
		objectDefinition1.getObjectDefinitionId());
}
else {
	objectDefinition2.setRootObjectDefinitionId(
		objectDefinition2.getObjectDefinitionId());
}
```

Now it is only: 
`objectDefinition.setRootObjectDefinitionId(objectDefinition1.getRootObjectDefinitionId());`

I think it should be:
```
if (objectDefinition1.isApproved() == objectDefinition.isApproved()) {
	objectDefinition.setRootObjectDefinitionId(
		objectDefinition1.getRootObjectDefinitionId());
}
else {
	objectDefinition.setRootObjectDefinitionId(
		objectDefinition.getObjectDefinitionId());
}
```